src/errors.rs

```diff
@@ -0,0 +1,123 @@
+use crate::{
+    protos::coresdk::activity_result::ActivityResult,
+    protos::coresdk::workflow_completion::WfActivationCompletion,
+    protos::temporal::api::workflowservice::v1::PollWorkflowTaskQueueResponse,
+    workflow::WorkflowError,
+};
+use tonic::codegen::http::uri::InvalidUri;
+
+pub(crate) struct ShutdownErr;
+pub(crate) struct WorkflowUpdateError {
+    /// Underlying workflow error
+    pub source: WorkflowError,
+    /// The run id of the erring workflow
+    pub run_id: String,
+}
+
+/// Errors thrown during initialization of [crate::Core]
+#[derive(thiserror::Error, Debug, displaydoc::Display)]
+pub enum CoreInitError {
+    /// Invalid URI: {0:?}
+    InvalidUri(#[from] InvalidUri),
+    /// Server connection error: {0:?}
+    TonicTransportError(#[from] tonic::transport::Error),
+}
+
+/// Errors thrown by [crate::Core::poll_workflow_task]
+#[derive(thiserror::Error, Debug, displaydoc::Display)]
+pub enum PollWfError {
+    /// There was an error specific to a workflow instance with id ({run_id}): {source:?}
+    WorkflowUpdateError {
+        /// Underlying workflow error
+        source: WorkflowError,
+        /// The run id of the erring workflow
+        run_id: String,
+    },
+    /// Poll workflow response from server was malformed: {0:?}
+    BadPollResponseFromServer(PollWorkflowTaskQueueResponse),
```

Comment:
Is this fatal? What should we do in this case? Doesn't seem like we can recover from this..

Reply:
Not necessarily. The server maybe just burps or something. This one is TBD